qutebrowser/javascript/webengine_caret.js

```diff
@@ -0,0 +1,1401 @@
+                (direction === "forward" ||
+                    direction === "right") &&
+                granularity === "word") {
+            CaretBrowsing.move("left", "character");
+        } else {
+            window.setTimeout(() => {
+                CaretBrowsing.updateCaretOrSelection(true);
+            }, 0);
+        }
+    };
+
+    CaretBrowsing.moveToBlock = function(paragraph, boundary) {
+        let action = "move";
+        if (CaretBrowsing.selectionEnabled) {
+            action = "extend";
+        }
+        window.
+            getSelection().
+            modify(action, paragraph, "paragraph");
+
+        window.
+            getSelection().
+            modify(action, boundary, "paragraphboundary");
+
+        window.setTimeout(() => {
+            CaretBrowsing.updateCaretOrSelection(true);
+        }, 0);
+    };
+
+    CaretBrowsing.toggle = function(value) {
+        if (CaretBrowsing.forceEnabled) {
+            CaretBrowsing.recreateCaretElement();
+            return;
+        }
+
+        if (value === undefined) {
+            CaretBrowsing.isEnabled = !CaretBrowsing.isEnabled;
+        } else {
+            CaretBrowsing.isEnabled = value;
+        }
+        CaretBrowsing.updateIsCaretVisible();
+    };
+
+    /**
+     * Event handler, called when the mouse is clicked. Chrome already
+     * sets the selection when the mouse is clicked, all we need to do is
+     * update our cursor.
+     * @param {Event} evt The DOM event.
+     * @return {boolean} True if the default action should be performed.
+     */
+    CaretBrowsing.onClick = function() {
+        if (!CaretBrowsing.isEnabled) {
+            return true;
+        }
+        window.setTimeout(() => {
+            CaretBrowsing.targetX = null;
+            CaretBrowsing.updateCaretOrSelection(false);
+        }, 0);
+        return true;
+    };
+
+    /**
+     * Called at a regular interval. Blink the cursor by changing its visibility.
+     */
+    CaretBrowsing.caretBlinkFunction = function() {
+        if (CaretBrowsing.caretElement) {
+            if (CaretBrowsing.blinkFlag) {
+                CaretBrowsing.caretElement.style.backgroundColor =
+                    CaretBrowsing.caretForeground;
+                CaretBrowsing.blinkFlag = false;
+            } else {
+                CaretBrowsing.caretElement.style.backgroundColor =
+                    CaretBrowsing.caretBackground;
+                CaretBrowsing.blinkFlag = true;
+            }
+        }
+    };
+
+    /**
+     * Update whether or not the caret is visible, based on whether caret browsing
+     * is enabled and whether this window / iframe has focus.
+     */
+    CaretBrowsing.updateIsCaretVisible = function() {
+        CaretBrowsing.isCaretVisible =
+            (CaretBrowsing.isEnabled && CaretBrowsing.isWindowFocused);
+        if (CaretBrowsing.isCaretVisible && !CaretBrowsing.caretElement) {
+            CaretBrowsing.setInitialCursor(CaretBrowsing.isWindows);
```

Comment:
This seems wrong - you assume a platform string in `setInitialCursor` but pass a boolean here. Instead of doing this, it might be better to have a separate `funcs.setPlatform` function which only does that, and nothing else?